XMonad/Hooks/EwmhDesktops.hs

```diff
@@ -171,10 +179,17 @@ ewmhDesktopsLogHookCustom t = withWindowSet $ \s -> do
         setNumberOfDesktops (length desktopNames)
         setDesktopNames desktopNames
 
-    -- Set client list; all windows, with focused windows last
-    let clientList = nub . concatMap (maybe [] (\(W.Stack x l r) -> reverse l ++ r ++ [x]) . W.stack) $ ws
+    -- Set client list which should be sorted by window age. We just
+    -- guess that StackSet contains windows list in this order which
+    -- isn't true but at least gives consistency with windows cycling
+    let clientList = nub . concatMap (W.integrate' . W.stack) $ ws
     whenChanged (ClientList clientList) $ setClientList clientList
 
+    -- Set stacking client list which should have bottom-to-top
+    -- stacking order, i.e. focused window should be last
```

Comment:
Note that focused window being last is not universally true. It's done by LayoutModifiers such as Magnifier that need it, and also by some layouts like MultiColumns (for reasons unknown), but most other layouts simply put the master window to the top.

Reply:
Yeah, but this comment means that in the result list focused window should be the last. And we assume here that focused window is the window that's selected in a `Stack` struct. It might be not always true, but it's what we have now, and I don't have a good example how it breaks something. So I left it as is because fixing it would require much more effort with no proper goal to reach :)

Reply:
Yeah, no action required, just wanted to note that in case it brings any forgotten memories out of anyone. :-)

Reply:
@liskin so I assume we can merge this then?  More time to see whether people complain :)

Reply:
Yeah, let's do it.